File: Sorts/mergeSort.js
const sort = (arr1, arr2) => {
  let tempArr = [];
  while ( arr1.length > 0 && arr2.length > 0 ) {
    if ( arr1[0] < arr2[0] ) {
      tempArr.push(arr1.shift());
    }
    else {
      tempArr.push(arr2.shift());
    }
  }
  return tempArr.concat(arr1).concat(arr2);
};

const mergeSort = (arr) => {
  if ( arr.length <= 1 ) return arr;

  const mid = Math.floor(arr.length / 2);
  const left = arr.slice(0,mid);
  const right = arr.slice(mid,arr.length);

  return sort( mergeSort(left), mergeSort(right) );
};

const sortMe = [5, -10, 20, 100, 0.1, 1, 20, 30, -100, 10000, 400, 20, 45, -74];

console.log(mergeSort(sortMe));

File: Sorts/merge_sort.py
def sort(arr1, arr2):
    tempArr = []

    while ( len(arr1) > 0 and len(arr2) > 0 ):
        if ( arr1[0] < arr2[0] ):
            tempArr.append(arr1.pop(0))
        else:
            tempArr.append(arr2.pop(0))
    return tempArr + arr1 + arr2


def mergeSort(arr):
    if ( len(arr) <= 1 ):
        return arr
    
    mid = len(arr) // 2
    left = arr[:mid]
    right = arr[mid:]

    return sort( mergeSort(left), mergeSort(right) )


print(mergeSort([2, 5, -10, 20, 5, 50,1, 600, -0.1]))